spec/oauth2/provider_spec.rb

```diff
@@ -19,18 +19,24 @@
   describe "access grant request" do
     shared_examples_for "asks for user permission" do
       it "creates an authorization" do
-        auth = mock_request(OAuth2::Provider::Authorization, :client => @client, :params => {}, :scopes => [], :valid? => true)
-        OAuth2::Provider::Authorization.should_receive(:new).with(@owner, params, nil).and_return(auth)
-        get(params)
+        # TODO: solve this mocking
```

Comment:
This needs to be resolved. Also I'd like to see some of the deleted/commented specs in this file restored. Any that are no longer needed should be deleted rather than commented out.